test/fake_device_factory_test.rb

```diff
@@ -21,11 +21,62 @@
 # find current contact information at www.suse.com.
 
 require_relative "spec_helper"
-require "storage"
-require "y2storage/fake_device_factory"
 
 describe Y2Storage::FakeDeviceFactory do
 
+  it "reads yaml of simple dasd and partition setup" do
+
+    environment = Storage::Environment.new(true, Storage::ProbeMode_NONE, Storage::TargetMode_DIRECT)
+    storage = Storage::Storage.new(environment)
+    staging = storage.staging
+
+    # rubocop:disable Style/StringLiterals
+
+    input = ['---',
+             '- dasd:',
+             '    name: "/dev/sda"',
+             '    size: 256 GiB',
+             '    block_size: 4 KiB',
+             '    partition_table: dasd',
```

Comment:
Yes, that line is needed. In fact, I think it is always needed. Why do you think it is not? Right now I think it is not possible to create a partition table without this line. A different thing is to create a Disk with a direct filesystem over it.

Regarding the syntax errors, the current implementation accepts a filesystem directly over the disk, or even to use the disk as PV or MD. I would not consider these options as a syntax error only because the library it is not prepared yet for that. In my opinion, the yaml reader only should be adapted if we are totally sure that these options are not possible.